src/amo/containers/App.js

```diff
@@ -77,7 +85,7 @@ export class AppBase extends React.Component {
           SearchFormComponent={SearchForm} isHomePage={isHomePage} location={location}
           query={query} ref={(ref) => { this.mastHead = ref; }} />
         <div className="App-content">
-          {children}
+          {errorPage.error ? 'errorContent' : children}
```

Comment:
You might be looking to change this later but you could move all the error logic into a generic `ErrorHandler` component and have it selectively render its children instead.

```jsx
// <App />
<div className="App-content">
  <ErrorHandler>
    {children}
  </ErrorHandler>
</div>

// <ErrorHandler />
render() {
  const { children, errorPage } = this.props;
  if (errorPage.error /* errorPage.lastError? */) {
    return <getErrorComponent(errorPage.status) />;
  }
  return children;
}
```